source/base/data_out_base.cc

```diff
@@ -2631,17 +2631,33 @@ namespace DataOutBase
         // used
         if (patch.reference_cell != ReferenceCells::get_hypercube<dim>())
           {
-            Point<spacedim> node;
+            const auto compute_node = [&](const unsigned int point_no) {
+              Point<spacedim> node;
+
+              if (patch.points_are_available)
+                {
+                  for (unsigned int d = 0; d < spacedim; ++d)
+                    node[d] =
+                      patch.data(patch.data.size(0) - spacedim + d, point_no);
+                  return node;
+                }
+              else
+                {
+                  AssertDimension(patch.n_subdivisions, 1);
+                  Assert(
+                    patch.reference_cell != ReferenceCells::Pyramid,
+                    ExcMessage(
+                      "Pyramids need different ordering of the vertices, which is not implemented yet here."));
+
+                  node = patch.vertices[point_no];
+                }
+
+              return node;
+            };
```

Comment:
I'd like to simplify the control flow here - it's hard to follow when we have a lambda named `compute_node` and a top-level function named `compute_node`.

Could we move the lambda into a new top-level function named `compute_simplex_node` and then rename `compute_node` to `compute_hypercube_node` or something similar? That would make this a lot easier to understand.

Reply:
Done!